routes.js

```diff
@@ -38,7 +42,35 @@ function fastifySwagger (fastify, opts, next) {
 
   // serve swagger-ui with the help of fastify-static
   fastify.register(require('fastify-static'), {
-    root: path.join(__dirname, 'static')
+    root: path.join(__dirname, 'static'),
+    prefix: staticPrefix
+  })
+
+  // Handler for external documentation files passed via $ref
+  fastify.route({
+    url: '/*',
+    method: 'GET',
+    schema: { hide: true },
+    handler: function (req, reply) {
+      const file = req.params['*']
+      if (file === '') {
+        reply.redirect(302, `${fastify.basePath}${staticPrefix}/index.html`)
+      } else if (opts.baseDir) {
+        const filePath = `${opts.baseDir}/${file}`
+
+        const ext = path.extname(file).toLocaleLowerCase()
+        if (!fs.existsSync(filePath) || ['.yaml', '.json'].indexOf(ext) === -1) {
+          return reply.code(404).send()
+        }
+
+        const contentType = ext === '.yaml' ? 'application/x-yaml' : 'application/json'
+        reply
+          .type(contentType)
+          .send(fs.readFileSync(filePath))
```

Comment:
I think it's better if you switch to fastify-static `reply.sendFile()` method. You can just add the decorator by passing `{ serve: false }` when registering fastify-static.

You'll need to register two instances of fastify-static, one with `{ serve: false }` to use inside this function, and one with `{ decorateReply: false }` to serve the swagger-ui. The two instances should not conflict/error. 

Reply:
Done